### PR TITLE
Rebuilt plan creation in admin

### PR DESCRIPTION
**Pull request by HancilSequeira**

### CODE REVIEW

File: Block/Adminhtml/Plan/Button/Back.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?php
+
```
Comment (sofiya02) [inline Block/Adminhtml/Plan/Button/Back.php:2]: too much space in file

File: Block/Adminhtml/Plan/Button/Save.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    /**
+     * Get URL for back
+     *
+     * @return string
+     */
+    private function getBackUrl()
+    {
+
```
Comment (sofiya02) [inline Block/Adminhtml/Plan/Button/Save.php:50]: unnecessary space

File: Block/Adminhtml/Plan/Edit/Button/Generic.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    /**
+     * @param Context $context
+     */
+    public function __construct(Context $context)
+    {
+        $this->context = $context;
+    }
+
+    /**
+     * @return mixed
```
Comment (sofiya02) [inline Block/Adminhtml/Plan/Edit/Button/Generic.php:21]: @return mixed ??

File: Model/Plans.php [hunk 4]
```diff
@@ -44,12 +44,15 @@ protected function _construct()
...
         return 1;
     }
- 
+
     public function getDisableStatus() {
         return 0;
     }
- 
+
     public function getAvailableStatuses() {
```
Comment (sofiya02) [inline Model/Plans.php:52]: remove unused functions from the file

File: Block/Adminhtml/Plan/Edit/Button/Generic.php (hunk 3 above)
Comment (HancilSequeira) [reply Block/Adminhtml/Plan/Edit/Button/Generic.php:21]: by default provided by the comment doc

File: Model/Plans.php (hunk 4 above)
Comment (HancilSequeira) [reply Model/Plans.php:52]: All this functions are used

**Review by sofiya02 — COMMENTED** [review]
- remove ds_store file
-  add "  <item name="sorting" xsi:type="string">desc</item>" in plan_listing.xml  line 304 and subscribed_listing.xml like 109 for desc order
- UI issue when showing list of plan for particular product

**Comment by sonyy-m** [comment]
Add .gitignore to ignore .DS_Store files

File: Block/Adminhtml/Plan/Edit/Form.php [hunk 5]
```diff
@@ -1,203 +0,0 @@
...
-                    array(
-                        'value'     => '',
-                        'label'     => 'Please Select',      
-                            ),
-                   array(
-                    'value'     => 'daily',
-                    'label'     => 'Daily',      
-                            ),
-                    array(
-                         'value'     => 'weekly',
```
[the remark this answers is not in the document]
Comment (HancilSequeira) [reply Block/Adminhtml/Plan/Edit/Form.php:136]: This code is removed

**Review by sofiya02 — APPROVED** [review]
LGTM

**Pull request merged**